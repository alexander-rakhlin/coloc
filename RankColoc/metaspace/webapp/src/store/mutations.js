import {isEqual, omit, pull, without} from 'lodash-es';
import router from '../router';
import compare from '../lib/compare';

import {
  decodeParams,
  encodeParams,
  encodeSections,
  encodeSortOrder,
  FILTER_SPECIFICATIONS,
  getFilterInitialValue,
  stripFilteringParams,
} from '../modules/Filters';
import {DEFAULT_ANNOTATION_VIEW_SECTIONS, DEFAULT_COLORMAP, DEFAULT_TABLE_ORDER} from '../modules/Filters/url';


function updatedLocation(state, filter) {
  let query = encodeParams(filter, state.route.path, state.filterLists);

  state.lastUsedFilters[state.route.path] = {
    filter,
    query,
    order: state.orderedActiveFilters
  };

  return {
    query: Object.assign(query, stripFilteringParams(state.route.query))
  };
}

function replaceURL(state, filter) {
  router.replace(updatedLocation(state, filter));
}

function pushURL(state, filter) {
  router.push(updatedLocation(state, filter));
}

function sortFilterKeys(keys) {
  keys.sort((a, b) => compare(FILTER_SPECIFICATIONS[a].sortOrder || 100, FILTER_SPECIFICATIONS[b].sortOrder || 100));
}

function updateFilter(state, filter, routerAction = null) {
  let active = [];

  // drop unset filters
  for (var i = 0; i < state.orderedActiveFilters.length; i++) {
    let key = state.orderedActiveFilters[i];
    if (filter[key] !== undefined)
      active.push(key);
  }

  // append newly added filters to the end
  for (var key in filter)
    if (filter[key] !== undefined &&
      active.indexOf(key) == -1)
      active.push(key);

  // sort
  sortFilterKeys(active);

  const changedFilterSet = !isEqual(state.orderedActiveFilters, active);

  state.orderedActiveFilters = active;
  if (routerAction === 'push' || (routerAction == null && changedFilterSet))
    pushURL(state, filter);
  else
    replaceURL(state, filter);
}

export default {
  updateFilter (state, filter) {
    updateFilter(state, filter);
  },

  replaceFilter (state, filter) {
    updateFilter(state, filter, 'replace');
  },

  addFilter (state, name) {
    const initialValue = getFilterInitialValue(name, state.filterLists);

    // FIXME: is there any way to access getters here?
    let filter = Object.assign(decodeParams(state.route, state.filterLists),
                               {[name]: initialValue});

    state.orderedActiveFilters.push(name);
    pushURL(state, filter);
  },

  updateFilterOnNavigate(state, to) {
    const newActiveFilters = Object.keys(decodeParams(to, state.filterLists));

    const removed = without(state.orderedActiveFilters, ...newActiveFilters);
    const added = without(newActiveFilters, ...state.orderedActiveFilters);

    pull(state.orderedActiveFilters, ...removed);
    state.orderedActiveFilters.push(...added);
    sortFilterKeys(state.orderedActiveFilters);
  },

  setFilterListsLoading(state) {
    state.filterListsLoading = true;
  },

  setFilterLists(state, filterLists) {
    state.filterLists = filterLists;
    state.filterListsLoading = false;
  },

  setAnnotation(state, annotation) {
    state.annotation = annotation;
  },

  updateAnnotationTableStatus(state, isLoading) {
    state.tableIsLoading = isLoading;
  },

  startTour(state, tourDescription) {
    state.currentTour = tourDescription;
  },

  endTour(state) {
    state.currentTour = null;
  },

  updateAnnotationViewSections(state, activeSections) {
    const sections = encodeSections(activeSections);
    const defaultSections = encodeSections(DEFAULT_ANNOTATION_VIEW_SECTIONS);
    router.replace({
      query: sections !== defaultSections
        ? { ...state.route.query, sections }
        : omit(state.route.query, 'sections'),
    });
  },

  setColormap(state, cmap) {
    router.replace({
      query: cmap !== DEFAULT_COLORMAP
        ? {...state.route.query, cmap}
        : omit(state.route.query, 'cmap'),
    });
  },

  setCurrentPage(state, page) {
    router.replace({
      query: page !== 1
        ? { ...state.route.query, page }
        : omit(state.route.query, 'page'),
    });
  },

  setSortOrder(state, sortOrder) {
    const sort = encodeSortOrder(sortOrder);
    const defaultSort = encodeSortOrder(DEFAULT_TABLE_ORDER);
    router.replace({
      query: sort !== defaultSort
        ? { ...state.route.query, sort }
        : omit(state.route.query, 'sort'),
    });
  },

  setDatasetTab(state, tab) {
    const path = {
      'list': '/datasets',
      'summary': '/datasets/summary'
    }[tab.toLowerCase()];
    router.replace({
      path,
      query: state.route.query,
    });
  }
};
