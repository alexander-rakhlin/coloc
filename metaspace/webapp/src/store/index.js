import Vue from 'vue';
import Vuex from 'vuex';
Vue.use(Vuex);

import getters from './getters.js';
import mutations from './mutations.js';
import actions from './actions.js';
import {accountModule} from '../modules/Account';

const store = new Vuex.Store({
  state: {
    // names of currently shown filters
    orderedActiveFilters: [],

    filterLists: null,
    filterListsLoading: false,

    // currently selected annotation
    annotation: undefined,

    // is annotation table loading?
    tableIsLoading: true,

    lastUsedFilters: {},

    currentTour: null,
  },

  getters,
  mutations,
  actions,
  modules: {
    account: accountModule
  }
})

export default store;
