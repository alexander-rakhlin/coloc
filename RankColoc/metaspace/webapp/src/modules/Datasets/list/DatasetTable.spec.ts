import { mount, config as testConfig } from '@vue/test-utils';
import DatasetTable from './DatasetTable.vue';
import router from '../../../router';
import { initMockGraphqlClient, provide } from '../../../../tests/utils/mockGraphqlClient';
import store from '../../../store/index';
import Vue from 'vue';
import Vuex from 'vuex';
import { sync } from 'vuex-router-sync';
jest.mock('file-saver');
import * as FileSaver from 'file-saver';
const mockFileSaver = FileSaver as jest.Mocked<typeof FileSaver>;

Vue.use(Vuex);
sync(store, router);

const blobToText = (blob: Blob) => new Promise<string>((resolve, reject) => {
  const reader = new FileReader();
  reader.addEventListener('load', () => { resolve(reader.result) });
  reader.addEventListener('error', (e) => { reject((e as any).error) });
  reader.readAsText(blob);
});

describe('DatasetTable', () => {
  const mockMetadataJson = JSON.stringify({
    Metadata_Type: "Imaging MS",
    MS_Analysis: {
      Detector_Resolving_Power: {mz: 1234, Resolving_Power: 123456}
    },
  });
  const mockFdrCounts = {
    dbName: 'HMDB-v2.5',
    levels: [0.05, 0.1, 0.5],
    counts: [1, 2, 20],
  };
  const mockDataset = {
    id: 'REPLACEME',
    status: 'FINISHED',
    metadataJson: mockMetadataJson,
    molDBs: ['HMDB-v2.5', 'HMDB-v4', 'CHEBI'],
    polarity: 'POSITIVE',
    fdrCounts: mockFdrCounts,
    groupApproved: true,
  };

  afterEach(() => {
    jest.useRealTimers();
  });

  it('should match snapshot', async () => {
    initMockGraphqlClient({
      Query: () => ({
        allDatasets: (_: any, {filter:{status}}: any) => {
          return [{...mockDataset, id: `${status}1`, status}]
        },
        countDatasets: () => 1,
      })
    });
    const wrapper = mount(DatasetTable, { store, router, provide, sync: false });
    await wrapper.vm.$data.loadingPromise;
    await Vue.nextTick();

    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to export a CSV', async () => {
    initMockGraphqlClient({
      Query: () => ({
        allDatasets: (_: any, {filter:{status}, offset}: any) => {
          return [
            {...mockDataset, id: `${status}${offset+1}`, status},
            {...mockDataset, groupApproved: false, id: `${status}${offset+2}`, status}
          ]
        },
        countDatasets: () => 4,
      })
    });
    const wrapper = mount(DatasetTable, { store, router, provide, sync: false });
    wrapper.setData({csvChunkSize: 2});
    await Vue.nextTick();

    await (wrapper.vm as any).startExport();
    expect(mockFileSaver.saveAs).toBeCalled();
    const blob: Blob = mockFileSaver.saveAs.mock.calls[0][0];
    const csv = await blobToText(blob);
    const csvWithoutDateHeader = csv.replace(/# Generated at .*\n/, '');

    expect(csvWithoutDateHeader).toMatchSnapshot();
  });
});
