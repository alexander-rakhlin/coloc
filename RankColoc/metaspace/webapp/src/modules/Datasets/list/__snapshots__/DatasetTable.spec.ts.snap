// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetTable should be able to export a CSV 1`] = `
"# URL: http://localhost/
\\"datasetId\\",\\"datasetName\\",\\"group\\",\\"submitter\\",\\"PI\\",\\"organism\\",\\"organismPart\\",\\"condition\\",\\"growthConditions\\",\\"ionisationSource\\",\\"maldiMatrix\\",\\"analyzer\\",\\"resPower400\\",\\"polarity\\",\\"uploadDateTime\\",\\"FDR@10% + DataBase\\",\\"opticalImage\\"
\\"FINISHED1\\",\\"datasets.0.name\\",\\"datasets.0.group.shortName\\",\\"datasets.0.submitter.name\\",\\"datasets.0.principalInvestigator.name\\",\\"datasets.0.organism\\",\\"datasets.0.organismPart\\",\\"datasets.0.condition\\",\\"datasets.0.growthConditions\\",\\"datasets.0.ionisationSource\\",\\"datasets.0.maldiMatrix\\",\\"datasets.0.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.0.uploadDateTime\\",\\"1,2,20 HMDB-v2.5\\",\\"http://localhostdatasets.0.rawOpticalImageUrl\\"
\\"FINISHED2\\",\\"datasets.1.name\\",\\"\\",\\"datasets.1.submitter.name\\",\\"datasets.1.principalInvestigator.name\\",\\"datasets.1.organism\\",\\"datasets.1.organismPart\\",\\"datasets.1.condition\\",\\"datasets.1.growthConditions\\",\\"datasets.1.ionisationSource\\",\\"datasets.1.maldiMatrix\\",\\"datasets.1.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.1.uploadDateTime\\",\\"1,2,20 HMDB-v2.5\\",\\"http://localhostdatasets.1.rawOpticalImageUrl\\"
\\"FINISHED3\\",\\"datasets.0.name\\",\\"datasets.0.group.shortName\\",\\"datasets.0.submitter.name\\",\\"datasets.0.principalInvestigator.name\\",\\"datasets.0.organism\\",\\"datasets.0.organismPart\\",\\"datasets.0.condition\\",\\"datasets.0.growthConditions\\",\\"datasets.0.ionisationSource\\",\\"datasets.0.maldiMatrix\\",\\"datasets.0.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.0.uploadDateTime\\",\\"1,2,20 HMDB-v2.5\\",\\"http://localhostdatasets.0.rawOpticalImageUrl\\"
\\"FINISHED4\\",\\"datasets.1.name\\",\\"\\",\\"datasets.1.submitter.name\\",\\"datasets.1.principalInvestigator.name\\",\\"datasets.1.organism\\",\\"datasets.1.organismPart\\",\\"datasets.1.condition\\",\\"datasets.1.growthConditions\\",\\"datasets.1.ionisationSource\\",\\"datasets.1.maldiMatrix\\",\\"datasets.1.analyzer.type\\",\\"13\\",\\"positive\\",\\"datasets.1.uploadDateTime\\",\\"1,2,20 HMDB-v2.5\\",\\"http://localhostdatasets.1.rawOpticalImageUrl\\"
"
`;

exports[`DatasetTable should match snapshot 1`] = `
<div>
  <div class="filter-panel">
    <mock-el-select placeholder="Add filter" class="filter-select">
      <mock-el-option value="group" label="Select group"></mock-el-option>
      <mock-el-option value="project" label="Select project"></mock-el-option>
      <mock-el-option value="submitter" label="Select submitter"></mock-el-option>
      <mock-el-option value="datasetIds" label="Select dataset"></mock-el-option>
      <mock-el-option value="polarity" label="Select polarity"></mock-el-option>
      <mock-el-option value="organism" label="Select organism"></mock-el-option>
      <mock-el-option value="organismPart" label="Select organism part"></mock-el-option>
      <mock-el-option value="condition" label="Select organism condition"></mock-el-option>
      <mock-el-option value="growthConditions" label="Select sample growth conditions"></mock-el-option>
      <mock-el-option value="analyzerType" label="Select analyzer"></mock-el-option>
      <mock-el-option value="ionisationSource" label="Select ionisation source"></mock-el-option>
      <mock-el-option value="maldiMatrix" label="Select MALDI matrix"></mock-el-option>
      <mock-el-option value="simpleQuery" label="Search anything"></mock-el-option>
      <mock-el-option value="metadataType" label="Select data type"></mock-el-option>
    </mock-el-select>
  </div>
  <div>
    <form class="el-form el-form--inline" style="display: inline-flex;">
      <div role="radiogroup" class="el-radio-group">
        <label role="radio" aria-checked="true" tabindex="0" class="el-radio-button el-radio-button--small is-active">
          <input type="radio" tabindex="-1" class="el-radio-button__orig-radio" value="List"><span class="el-radio-button__inner">List</span></label>
        <label role="radio" tabindex="-1" class="el-radio-button el-radio-button--small">
          <input type="radio" tabindex="-1" class="el-radio-button__orig-radio" value="Summary"><span class="el-radio-button__inner">Summary</span></label>
      </div>
      <div role="group" aria-label="checkbox-group" class="el-checkbox-group" style="padding: 5px 20px;">
        <label role="checkbox" aria-checked="true" class="el-checkbox cb-started is-checked"><span aria-checked="mixed" class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span>
          <input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="started">
          </span><span class="el-checkbox__label">Processing (1)<!----></span></label>
        <label role="checkbox" aria-checked="true" class="el-checkbox cb-queued is-checked"><span aria-checked="mixed" class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span>
          <input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="queued">
          </span><span class="el-checkbox__label">Queued (1)<!----></span></label>
        <label role="checkbox" aria-checked="true" class="el-checkbox is-checked"><span aria-checked="mixed" class="el-checkbox__input is-checked"><span class="el-checkbox__inner"></span>
          <input type="checkbox" aria-hidden="true" class="el-checkbox__original" value="finished">
          </span><span class="el-checkbox__label">Finished (1)<!----></span></label>
        <!---->
      </div>
    </form>
  </div>
  <div>
    <div id="dataset-list-header">
      <div style="font: sans-serif 24px 24px; padding: 5px;"><span>
          Recent uploads
        </span></div>
      <button type="button" class="el-button export-btn el-button--default">
        <!---->
        <!----><span>
        Export to CSV
      </span></button>
    </div>
    <div class="dataset-list allow-double-column">
      <div class="dataset-item">
        <mock-el-dialog title="Provided metadata">
          <div class="el-row">
            <div role="tree" class="el-tree" id="metadata-tree">
              <div role="treeitem" tabindex="0" aria-expanded="true" draggable="false" class="el-tree-node is-expanded is-focusable">
                <div class="el-tree-node__content" style="padding-left: 0px;"><span class="el-tree-node__expand-icon el-icon-caret-right expanded"></span>
                  <!---->
                  <!----><span class="el-tree-node__label">MS analysis</span></div>
                <div role="group" aria-expanded="true" class="el-tree-node__children">
                  <div role="treeitem" tabindex="-1" aria-expanded="true" draggable="false" class="el-tree-node is-expanded is-focusable">
                    <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right expanded"></span>
                      <!---->
                      <!----><span class="el-tree-node__label">Detector resolving power</span></div>
                    <div role="group" aria-expanded="true" class="el-tree-node__children">
                      <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                        <div class="el-tree-node__content" style="padding-left: 36px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                          <!---->
                          <!----><span class="el-tree-node__label">mz: 1234</span></div>
                        <!---->
                      </div>
                      <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                        <div class="el-tree-node__content" style="padding-left: 36px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                          <!---->
                          <!----><span class="el-tree-node__label">Resolving power: 123456</span></div>
                        <!---->
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div role="treeitem" tabindex="-1" aria-expanded="true" draggable="false" class="el-tree-node is-expanded is-focusable">
                <div class="el-tree-node__content" style="padding-left: 0px;"><span class="el-tree-node__expand-icon el-icon-caret-right expanded"></span>
                  <!---->
                  <!----><span class="el-tree-node__label">Data Management</span></div>
                <div role="group" aria-expanded="true" class="el-tree-node__children">
                  <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                    <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                      <!---->
                      <!----><span class="el-tree-node__label">Submitter: allDatasets.0.submitter.name</span></div>
                    <!---->
                  </div>
                  <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                    <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                      <!---->
                      <!----><span class="el-tree-node__label">Principal Investigator: allDatasets.0.principalInvestigator.name</span></div>
                    <!---->
                  </div>
                  <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                    <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                      <!---->
                      <!----><span class="el-tree-node__label">Group: allDatasets.0.group.name</span></div>
                    <!---->
                  </div>
                  <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                    <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                      <!---->
                      <!----><span class="el-tree-node__label">Projects: allDatasets.0.projects.0.name, allDatasets.0.projects.1.name</span></div>
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <div class="el-tree__drop-indicator" style="display: none;"></div>
            </div>
          </div>
        </mock-el-dialog>
        <div class="opt-image">
          <div class="edit-opt-image-guest">
            <img src="allDatasets.0.thumbnailOpticalImageUrl" alt="Optical image">
          </div>
        </div>
        <div class="ds-info">
          <div class="ds-item-line">
            <b>allDatasets.0.name</b>
          </div>
          <div class="ds-item-line"><span title="Filter by species" class="ds-add-filter">
        allDatasets.0.organism</span>,
            <span title="Filter by organism part" class="ds-add-filter">
        alldatasets.0.organismpart</span> <span title="Filter by condition" class="ds-add-filter">
        (alldatasets.0.condition)</span></div>
          <div class="ds-item-line"><span title="Filter by ionisation source" class="ds-add-filter">
        allDatasets.0.ionisationSource</span> +
            <span title="Filter by analyzer type" class="ds-add-filter">
        allDatasets.0.analyzer.type</span>,
            <span title="Filter by polarity" class="ds-add-filter">
        positive mode</span>, RP 123k @ 1234
          </div>
          <div class="ds-item-line" style="font-size: 15px;">
            Submitted <span class="s-bold">????-??-??</span> at ??:?? by
            <span title="Filter by submitter" class="ds-add-filter">
        allDatasets.0.submitter.name</span><span>,
        <mock-el-dropdown showtimeout="50" placement="bottom" trigger="hover"><div slot-key="default"><span class="s-group ds-add-filter">
            allDatasets.0.group.shortName
          </span> </div>
          <div slot-key="dropdown">
            <mock-el-dropdown-menu>
              <mock-el-dropdown-item command="filter_group">Filter by this group</mock-el-dropdown-item>
              <mock-el-dropdown-item command="view_group">View group</mock-el-dropdown-item>
            </mock-el-dropdown-menu>
          </div>
          </mock-el-dropdown>
          </span>
        </div>
        <!---->
      </div>
      <div class="ds-actions">
        <!----><span><div title="Processing is under way" class="striped-progressbar processing"></div></span>
        <!---->
        <i class="el-icon-view"></i>
        <a class="metadata-link">Show full metadata</a>
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
    </div>
    <div class="dataset-item odd">
      <mock-el-dialog title="Provided metadata">
        <div class="el-row">
          <div role="tree" class="el-tree" id="metadata-tree">
            <div role="treeitem" tabindex="0" aria-expanded="true" draggable="false" class="el-tree-node is-expanded is-focusable">
              <div class="el-tree-node__content" style="padding-left: 0px;"><span class="el-tree-node__expand-icon el-icon-caret-right expanded"></span>
                <!---->
                <!----><span class="el-tree-node__label">MS analysis</span></div>
              <div role="group" aria-expanded="true" class="el-tree-node__children">
                <div role="treeitem" tabindex="-1" aria-expanded="true" draggable="false" class="el-tree-node is-expanded is-focusable">
                  <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right expanded"></span>
                    <!---->
                    <!----><span class="el-tree-node__label">Detector resolving power</span></div>
                  <div role="group" aria-expanded="true" class="el-tree-node__children">
                    <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                      <div class="el-tree-node__content" style="padding-left: 36px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                        <!---->
                        <!----><span class="el-tree-node__label">mz: 1234</span></div>
                      <!---->
                    </div>
                    <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                      <div class="el-tree-node__content" style="padding-left: 36px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                        <!---->
                        <!----><span class="el-tree-node__label">Resolving power: 123456</span></div>
                      <!---->
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div role="treeitem" tabindex="-1" aria-expanded="true" draggable="false" class="el-tree-node is-expanded is-focusable">
              <div class="el-tree-node__content" style="padding-left: 0px;"><span class="el-tree-node__expand-icon el-icon-caret-right expanded"></span>
                <!---->
                <!----><span class="el-tree-node__label">Data Management</span></div>
              <div role="group" aria-expanded="true" class="el-tree-node__children">
                <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                  <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                    <!---->
                    <!----><span class="el-tree-node__label">Submitter: allDatasets.0.submitter.name</span></div>
                  <!---->
                </div>
                <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                  <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                    <!---->
                    <!----><span class="el-tree-node__label">Principal Investigator: allDatasets.0.principalInvestigator.name</span></div>
                  <!---->
                </div>
                <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                  <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                    <!---->
                    <!----><span class="el-tree-node__label">Group: allDatasets.0.group.name</span></div>
                  <!---->
                </div>
                <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                  <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                    <!---->
                    <!----><span class="el-tree-node__label">Projects: allDatasets.0.projects.0.name, allDatasets.0.projects.1.name</span></div>
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <div class="el-tree__drop-indicator" style="display: none;"></div>
          </div>
        </div>
      </mock-el-dialog>
      <div class="opt-image">
        <div class="edit-opt-image-guest">
          <img src="allDatasets.0.thumbnailOpticalImageUrl" alt="Optical image">
        </div>
      </div>
      <div class="ds-info">
        <div class="ds-item-line">
          <b>allDatasets.0.name</b>
        </div>
        <div class="ds-item-line"><span title="Filter by species" class="ds-add-filter">
        allDatasets.0.organism</span>,
          <span title="Filter by organism part" class="ds-add-filter">
        alldatasets.0.organismpart</span> <span title="Filter by condition" class="ds-add-filter">
        (alldatasets.0.condition)</span></div>
        <div class="ds-item-line"><span title="Filter by ionisation source" class="ds-add-filter">
        allDatasets.0.ionisationSource</span> +
          <span title="Filter by analyzer type" class="ds-add-filter">
        allDatasets.0.analyzer.type</span>,
          <span title="Filter by polarity" class="ds-add-filter">
        positive mode</span>, RP 123k @ 1234
        </div>
        <div class="ds-item-line" style="font-size: 15px;">
          Submitted <span class="s-bold">????-??-??</span> at ??:?? by
          <span title="Filter by submitter" class="ds-add-filter">
        allDatasets.0.submitter.name</span><span>,
        <mock-el-dropdown showtimeout="50" placement="bottom" trigger="hover"><div slot-key="default"><span class="s-group ds-add-filter">
            allDatasets.0.group.shortName
          </span> </div>
        <div slot-key="dropdown">
          <mock-el-dropdown-menu>
            <mock-el-dropdown-item command="filter_group">Filter by this group</mock-el-dropdown-item>
            <mock-el-dropdown-item command="view_group">View group</mock-el-dropdown-item>
          </mock-el-dropdown-menu>
        </div>
        </mock-el-dropdown>
        </span>
      </div>
      <!---->
    </div>
    <div class="ds-actions">
      <!---->
      <!----><span><div title="Waiting in the queue" class="striped-progressbar queued"></div></span>
      <i class="el-icon-view"></i>
      <a class="metadata-link">Show full metadata</a>
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
  </div>
  <div class="dataset-item">
    <mock-el-dialog title="Provided metadata">
      <div class="el-row">
        <div role="tree" class="el-tree" id="metadata-tree">
          <div role="treeitem" tabindex="0" aria-expanded="true" draggable="false" class="el-tree-node is-expanded is-focusable">
            <div class="el-tree-node__content" style="padding-left: 0px;"><span class="el-tree-node__expand-icon el-icon-caret-right expanded"></span>
              <!---->
              <!----><span class="el-tree-node__label">MS analysis</span></div>
            <div role="group" aria-expanded="true" class="el-tree-node__children">
              <div role="treeitem" tabindex="-1" aria-expanded="true" draggable="false" class="el-tree-node is-expanded is-focusable">
                <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right expanded"></span>
                  <!---->
                  <!----><span class="el-tree-node__label">Detector resolving power</span></div>
                <div role="group" aria-expanded="true" class="el-tree-node__children">
                  <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                    <div class="el-tree-node__content" style="padding-left: 36px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                      <!---->
                      <!----><span class="el-tree-node__label">mz: 1234</span></div>
                    <!---->
                  </div>
                  <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                    <div class="el-tree-node__content" style="padding-left: 36px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                      <!---->
                      <!----><span class="el-tree-node__label">Resolving power: 123456</span></div>
                    <!---->
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div role="treeitem" tabindex="-1" aria-expanded="true" draggable="false" class="el-tree-node is-expanded is-focusable">
            <div class="el-tree-node__content" style="padding-left: 0px;"><span class="el-tree-node__expand-icon el-icon-caret-right expanded"></span>
              <!---->
              <!----><span class="el-tree-node__label">Data Management</span></div>
            <div role="group" aria-expanded="true" class="el-tree-node__children">
              <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                  <!---->
                  <!----><span class="el-tree-node__label">Submitter: allDatasets.0.submitter.name</span></div>
                <!---->
              </div>
              <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                  <!---->
                  <!----><span class="el-tree-node__label">Principal Investigator: allDatasets.0.principalInvestigator.name</span></div>
                <!---->
              </div>
              <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                  <!---->
                  <!----><span class="el-tree-node__label">Group: allDatasets.0.group.name</span></div>
                <!---->
              </div>
              <div role="treeitem" tabindex="-1" draggable="false" class="el-tree-node is-focusable">
                <div class="el-tree-node__content" style="padding-left: 18px;"><span class="el-tree-node__expand-icon el-icon-caret-right is-leaf"></span>
                  <!---->
                  <!----><span class="el-tree-node__label">Projects: allDatasets.0.projects.0.name, allDatasets.0.projects.1.name</span></div>
                <!---->
              </div>
            </div>
          </div>
          <!---->
          <div class="el-tree__drop-indicator" style="display: none;"></div>
        </div>
      </div>
    </mock-el-dialog>
    <div class="opt-image">
      <div class="edit-opt-image-guest">
        <img src="allDatasets.0.thumbnailOpticalImageUrl" alt="Optical image">
      </div>
    </div>
    <div class="ds-info">
      <div class="ds-item-line">
        <b>allDatasets.0.name</b>
      </div>
      <div class="ds-item-line"><span title="Filter by species" class="ds-add-filter">
        allDatasets.0.organism</span>,
        <span title="Filter by organism part" class="ds-add-filter">
        alldatasets.0.organismpart</span> <span title="Filter by condition" class="ds-add-filter">
        (alldatasets.0.condition)</span></div>
      <div class="ds-item-line"><span title="Filter by ionisation source" class="ds-add-filter">
        allDatasets.0.ionisationSource</span> +
        <span title="Filter by analyzer type" class="ds-add-filter">
        allDatasets.0.analyzer.type</span>,
        <span title="Filter by polarity" class="ds-add-filter">
        positive mode</span>, RP 123k @ 1234
      </div>
      <div class="ds-item-line" style="font-size: 15px;">
        Submitted <span class="s-bold">????-??-??</span> at ??:?? by
        <span title="Filter by submitter" class="ds-add-filter">
        allDatasets.0.submitter.name</span><span>,
        <mock-el-dropdown showtimeout="50" placement="bottom" trigger="hover"><div slot-key="default"><span class="s-group ds-add-filter">
            allDatasets.0.group.shortName
          </span> </div>
      <div slot-key="dropdown">
        <mock-el-dropdown-menu>
          <mock-el-dropdown-item command="filter_group">Filter by this group</mock-el-dropdown-item>
          <mock-el-dropdown-item command="view_group">View group</mock-el-dropdown-item>
        </mock-el-dropdown-menu>
      </div>
      </mock-el-dropdown>
      </span>
    </div>
    <div class="ds-item-line"><span><a href="/annotations?db=HMDB-v2.5&amp;ds=FINISHED1&amp;mdtype=allDatasets.0.metadataType" class="">1, 2, 20 annotations</a>
        @ FDR 0.05, 0.1, 0.5% (HMDB-v2.5)
      </span></div>
  </div>
  <div class="ds-actions"><span><i class="el-icon-picture"></i> <mock-el-popover trigger="hover" placement="top"><div slot-key="default"><div class="db-link-list">
          Select a database:
          <div><a href="/annotations?db=HMDB-v2.5&amp;ds=FINISHED1&amp;mdtype=allDatasets.0.metadataType" class="">
              HMDB-v2.5
            </a></div><div><a href="/annotations?db=HMDB-v4&amp;ds=FINISHED1&amp;mdtype=allDatasets.0.metadataType" class="">
              HMDB-v4
            </a></div><div><a href="/annotations?db=CHEBI&amp;ds=FINISHED1&amp;mdtype=allDatasets.0.metadataType" class="">
              CHEBI
            </a></div></div> </div><div slot-key="reference"><a>Browse annotations</a></div></mock-el-popover> <br></span>
    <!---->
    <!---->
    <i class="el-icon-view"></i>
    <a class="metadata-link">Show full metadata</a>
    <!---->
    <!---->
    <!---->
    <!---->
  </div>
</div>
</div>
</div>
</div>
`;
